### PR TITLE
Add CI to validate URLs and enhance Python code quality

### PR DESCRIPTION
**Pull request by jcwchen**
**Description**
- Add a CI for check-in PRs and weekly check
   - flake8 check for Python code and notebook
   - add validate_urls to validate all URLs under onnx/tutorials
   - Weekly check can early detect broken URLs
- Fix current flake failures
- Correct URLs
  - Update to newer URLs if any
  - Use https:// instead of http: for security
  - Remove retired/deprecated ones. For instance, MXNET Model Zoo
- Add pull_request_template.md to improve PR content

**Motivation and Context**
- A lot of old URLs are broken
- Improve code quality
- There is no CI to validate new documents/notebooks. Closes https://github.com/onnx/tutorials/issues/22.

### CODE REVIEW

**Review by AlexandreEichenberger — COMMENTED** [review]
Its great to have switched to the `https` links, and I like that there is a weekly cron job to make sure the links are not stale.

Would it make sense to also activate the checks as part of the CIs for a new PR? Apologies if that is already the case.

Also, some of the files, e.g. the workflow_scripts url validators, appear not to have a copyright statement. They should have their original copyrights, or the Apache-2 if there were created by the author of this PR.

**Comment by AlexandreEichenberger** [comment]
> Currently the URL check in CI only validates modifications (use git diff to get those modified files). I think it should be sufficient because other URL validation which is not modified by that PR should not block the PR. Therefore global URL check will only run weekly.

Does this cover the scenario where a valid URL in the doc becomes stale because the destination website modified their urls?

File: .github/workflows/validate.yml [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+    - name: Validate changed URLs from this PR
+      run: |
+        git fetch --all
+        python workflow_scripts/validate_changed_urls.py
+
+    - name: Weekly validate all URLs under onnx/tutorials
+      if: (github.event_name == 'schedule') # Only triggered by weekly event
+      run: |
+        python workflow_scripts/validate_all_urls.py
```
Comment (jcwchen) [inline .github/workflows/validate.yml:43]: FYI @AlexandreEichenberger

**Comment by jcwchen** [comment]
> Does this cover the scenario where a valid URL in the doc becomes stale because the destination website modified their urls?

Yes, such a case will be covered by the weekly check -- The CI will validate all URLs under onnx/tutorials to prevent stale URL.

**Review by AlexandreEichenberger — APPROVED** [review]
LGTM

**Pull request merged**